src/collector/dropper.rs

```diff
@@ -3,44 +3,47 @@ use std::sync::atomic::Ordering;
 use std::sync::Arc;
 use std::thread::spawn;
 
-use crossbeam::channel::{self, SendError, Sender};
-use parking_lot::RwLock;
-use rayon::iter::IntoParallelRefIterator;
-use rayon::iter::ParallelIterator;
+use crossbeam::channel::{self, Receiver, SendError, Sender};
 
 use crate::collector::GcData;
+use crate::concurrency::cross_thread_buffer::CrossThreadBuffer;
+
+type DropBuffer = CrossThreadBuffer<Arc<GcData>>;
 
 pub(crate) struct BackgroundDropper {
```

Comment:
The changes in here are pure premature optimization. But I think it is better than the previous design